molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/AttributeTypeServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+package org.molgenis.oneclickimporter.service.Impl;
+
+import org.molgenis.data.meta.AttributeType;
+import org.molgenis.oneclickimporter.service.AttributeTypeService;
+import org.molgenis.util.MolgenisDateFormat;
+import org.springframework.stereotype.Component;
+
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+import static org.molgenis.data.meta.AttributeType.*;
+
+@Component
+public class AttributeTypeServiceImpl implements AttributeTypeService
+{
+	private static final int MAX_STRING_LENGTH = 255;
+
+	@Override
+	public AttributeType guessAttributeType(List<Object> dataValues)
+	{
+		boolean guessCompleted = false;
+		int rowCount = dataValues.size();
+		int currentRowIndex = 0;
+
+		AttributeType guess = getBasicAttributeType(dataValues.get(0));
+		while (currentRowIndex < rowCount && !guessCompleted)
+		{
+			Object value = dataValues.get(currentRowIndex);
+			AttributeType basicType = getBasicAttributeType(value);
+
+			guess = getCommonType(guess, basicType);
+			guess = getEnrichedType(guess, value);
```

Comment:
why not create a new var enrichedGuess reusing the var can lead to problems when moving the lines around